src/components/Onboarding/steps.ts

```diff
@@ -0,0 +1,261 @@
+/* eslint-disable max-lines */
+import { Translate } from 'next-translate';
+import { Step as BaseStep } from 'react-joyride';
+
+// import OnboardingStep from './OnboardingStep';
+
+import OnboardingGroup from '@/types/OnboardingGroup';
+import { getSurahNavigationUrl } from '@/utils/navigation';
+
+type Step = BaseStep & {
+  showNextButton?: boolean;
+  showPrevButton?: boolean;
+};
+
+// eslint-disable-next-line react-func/max-lines-per-function
+export const checklistIndexToOnboardingSteps = (
+  t: Translate,
+  component: React.ElementType,
+): Record<
+  OnboardingGroup,
+  {
+    title: string;
+    description: string;
+    step: Partial<Step>;
+  }[]
+> => {
+  const commonStepOptions: Partial<Step> = {
+    content: '',
+    placement: 'auto',
+    disableBeacon: true,
+    hideBackButton: true,
+    disableOverlay: false,
+    disableOverlayClose: true,
+    tooltipComponent: component as any,
+    floaterProps: {
+      hideArrow: true,
+    },
+    spotlightPadding: 10,
+  };
+
+  const getDetails = (key: string) => ({
+    title: t(`onboarding:steps.${key}.title`),
+    description: t(`onboarding:steps.${key}.description`),
+  });
+
+  return {
+    [OnboardingGroup.HOMEPAGE]: [
+      {
+        ...getDetails('quran-radio'),
+        step: {
+          ...commonStepOptions,
+          target: '#radio-button',
+          showSkipButton: true,
+        },
+      },
+      {
+        ...getDetails('search-bar'),
+        step: {
+          ...commonStepOptions,
+          target: '#command-bar',
+        },
+      },
+      {
+        ...getDetails('voice-search'),
+        step: {
+          ...commonStepOptions,
+          target: '#voice-search-trigger',
+          placement: 'bottom',
+        },
+      },
+
+      {
+        ...getDetails('juz'),
+        step: {
+          ...commonStepOptions,
+          target: '#juz-tab',
+        },
+      },
+    ],
+    [OnboardingGroup.SETTINGS]: [
+      {
+        ...getDetails('settings'),
+        step: {
+          ...commonStepOptions,
+          target: '#settings-button',
+          showNextButton: false,
+          showSkipButton: true,
+          spotlightClicks: true,
+        },
+      },
+      {
+        ...getDetails('theme'),
+        step: {
+          ...commonStepOptions,
+          target: '#theme-section',
+          spotlightClicks: true,
+          placement: 'left-start',
+          disableScrolling: true,
+        },
+      },
+      {
+        ...getDetails('font-style'),
```

Comment:
There is an issue on mobile during the settings flow that prevents scrolling and exiting


https://github.com/quran/quran.com-frontend-next/assets/15169499/427e97e2-32a7-49a6-88c4-44dab4215a2c